generators/languages/index.js

```diff
@@ -34,57 +64,92 @@ module.exports = LanguagesGenerator.extend({
     prompting : function () {
         var cb = this.async();
 
+        var languagesArgument = this.languagesArgument;
+        var languageOptions = this.getAllSupportedLanguageOptions();
         var prompts = [
         {
+            when: function(){
+                return languagesArgument === undefined;
+            },
             type: 'checkbox',
             name: 'languages',
             message: 'Please choose additional languages to install',
-            choices: [
-                {name: 'Catalan', value: 'ca'},
-                {name: 'Chinese (Simplified)', value: 'zh-cn'},
-                {name: 'Chinese (Traditional)', value: 'zh-tw'},
-                {name: 'Danish', value: 'da'},
-                {name: 'Dutch', value: 'nl'},
-                {name: 'Galician', value: 'gl'},
-                {name: 'German', value: 'de'},
-                {name: 'Hungarian', value: 'hu'},
-                {name: 'Italian', value: 'it'},
-                {name: 'Japanese', value: 'ja'},
-                {name: 'Korean', value: 'ko'},
-                {name: 'Polish', value: 'pl'},
-                {name: 'Portuguese (Brazilian)', value: 'pt-br'},
-                {name: 'Portuguese', value: 'pt-pt'},
-                {name: 'Romanian', value: 'ro'},
-                {name: 'Russian', value: 'ru'},
-                {name: 'Spanish', value: 'es'},
-                {name: 'Swedish', value: 'sv'},
-                {name: 'Turkish', value: 'tr'},
-                {name: 'Tamil', value: 'ta'}
-            ],
-            default: 0
+            choices: languageOptions
         }];
-        if (this.enableTranslation) {
+        if (this.enableTranslation || configOptions.enableTranslation) {
             this.prompt(prompts, function (props) {
-                this.languages = props.languages;
+                this.languages = this.languagesArgument || props.languages;
                 cb();
             }.bind(this));
-        }else{
+        } else {
             this.log(chalk.red('Translation is disabled for the project. Language cannot be added.'));
             return;
         }
     },
 
+    default: {
+
+        getSharedConfigOptions: function () {
+            if(configOptions.applicationType) {
```

Comment:
Shouldn't we get only the variables that are required by language sub gen?  I mean the only ootions used in index.js or in language templates?